datatypes/src/jvmMain/kotlin/at/asitplus/crypto/datatypes/JcaExtensions.kt

```diff
@@ -130,8 +138,30 @@ val CryptoSignature.jcaSignatureBytes: ByteArray
 /**
  * In Java EC signatures are returned as DER-encoded, RSA signatures however are raw bytearrays
  */
-fun CryptoSignature.Companion.parseFromJca(input: ByteArray, algorithm: CryptoAlgorithm)=
+fun CryptoSignature.Companion.parseFromJca(input: ByteArray, algorithm: CryptoAlgorithm) =
     if (algorithm.isEc)
         CryptoSignature.decodeFromDer(input)
     else
         CryptoSignature.RSAorHMAC(input)
+
+/**
+ * Converts this [X509Certificate] to a [java.security.cert.X509Certificate].
+ * This function is suspending, because it uses a mutex to lock the underlying certificate factory (which is reused for performance reasons
+ */
+suspend fun X509Certificate.toJcaCertificate(): KmmResult<java.security.cert.X509Certificate> = runCatching {
+    crtFactMut.withLock {
```

Comment:
So this locking is necessary because the certificate factory is not thread safe?

Reply:
the BC one is mos def not thread-safe and the one from the Sun provider currently is and always has been, but may not be coroutine-safe. Anyways, no guarantees for future releases packaged into the java stdlib, so I am assuming it to be not thread-safe and not coroutine-safe